### PR TITLE
Resolve target Model primary key type for BelongsTo|HasManyKey|HasOnKey

### LINKED ISSUE #115: BelongsTo relationship generates wrong xID type when using UUID
Given:

``` go
Model("Identity", func() {
    Description("")
    Field("id", gorma.UUID, func() {
        PrimaryKey()
        SQLTag("type:uuid")
        Description("This is the ID PK field")
    })
    Field("full_name", gorma.String, func() {
        Description("The fullname of the Identity")
        Alias("fullName")
    })
    Field("image_url", gorma.String, func() {
        Description("The image URL for this Identity")
        Alias("imageURL")
    })
})
Model("User", func() {
    Description("")
    Field("id", gorma.UUID, func() {
        PrimaryKey()
                SQLTag("type:uuid")
        Description("This is the ID PK field")
    })
    Field("email", gorma.String, func() {
        Description("This is the unique email field")
    })
    BelongsTo("Identity")
})
```

Expected:

``` go
type Identity struct {
    ID        uuid.UUID `sql:"type:uuid" gorm:"primary_key"` // This is the ID PK field
        ...
    FullName  string `gorm:"column:fullName"` // The fullname of the Identity
    ImageURL  string `gorm:"column:imageURL"` // The image URL for this Identity
}

type User struct {
    ID        uuid.UUID `sql:"type:uuid" gorm:"primary_key"` // This is the ID PK field
       ...
    IdentityID uuid.UUID   // Belongs To Identity
    Identity   Identity
}
```

But got:

``` go
type User struct {
    ID        uuid.UUID `sql:"type:uuid" gorm:"primary_key"` // This is the ID PK field
       ...
    IdentityID int   // Belongs To Identity  // INT ??
    Identity   Identity
}
```

### PR DESCRIPTION
**Pull request by aslakknutsen**
Instead of hard-coding the Type of a BelongsTo|HasManyKey|HasOneKey relationship
to be in, resolve the target model in the relationship and use its PrimaryKey as
xID type in source struct.

Note: Currently this will panic if multiple PrimaryKeys are defined.

fixes #115

### CODE REVIEW

**Comment by aslakknutsen** [comment]
Just noticed, while the correct Go Type is now working, the SQLTag type is not. And SQLTag can contain anything, so I guess we need to parse it and look for 'only parts of it' that would be relevant to a foreign key. (e.g. type is, but default is not)

Maybe we need to open up this in the DSL? 

```go
BelongsTo("Identity", func() {
	SQLTag("type:uuid")
})
```

**Comment by aslakknutsen** [comment]
Also, BelongsTo filters currently not working. Hardcoded to int

File: relationalfield.go [hunk 1]
```diff
@@ -121,6 +123,51 @@ func goDatatype(f *RelationalFieldDefinition, includePtr bool) string {
...
+	}
+	return belongsToIDType(f, true)
+}
+
+func belongsToIDType(f *RelationalFieldDefinition, includePtr bool) string {
+	if f.Parent == nil {
+		return "int"
+	}
+	modelName := strings.Replace(f.FieldName, "ID", "", -1)
+	model := f.Parent.BelongsTo[modelName]
```
Comment (bketelsen) [inline relationalfield.go]: what happens when these two variables (modelName and model) come up empty?  Should there be error handling here?  same below.

File: relationalfield.go [hunk 2]
```diff
@@ -121,6 +123,51 @@ func goDatatype(f *RelationalFieldDefinition, includePtr bool) string {
...
+	if f.Parent == nil {
+		return "int"
+	}
+	modelName := strings.Replace(f.FieldName, "ID", "", -1)
+	model := f.Parent.HasMany[modelName]
+	return relatedIDType(model, includePtr)
+}
+
+func relatedIDType(m *RelationalModelDefinition, includePtr bool) string {
+	if m == nil {
```
Comment (bketelsen) [inline relationalfield.go]: OK I see if the model is nil you'll just default to an int.  That may work.

**Comment by bketelsen** [comment]
some questions in the code, generally I don't see a problem.  Have you generated an app with this yet?

**Comment by aslakknutsen** [comment]
Yes I have. This was what I had to do to get UUID working. :) It's currently semi depending on a change in Goa as well to reintroduce satori/go.uuid that has the Scan/Value interface implemented.

**Comment by aslakknutsen** [comment]
The current impl shortcuts a little and makes it "BelongsTo("Model", "SQLTag")' as oppose to passing in a dsl func to BelongsTo that supports SQLTag. What's your opinion here?

**Comment by nicholaswyoung** [comment]
I'm currently blocked on Gorma UUID support. Any idea when this PR might be merged?

**Pull request merged**